draftquality/feature_lists/ptwiki.py

```diff
@@ -0,0 +1,149 @@
+    wikitext.revision.break_chars / max(wikitext.revision.chars, 1),
+    wikitext.revision.longest_repeated_char / max(wikitext.revision.chars, 1)
+]
+
+token_based = [
+    wikitext.revision.tokens,
+    wikitext.revision.numbers,
+    wikitext.revision.whitespaces,
+    wikitext.revision.markups,
+    wikitext.revision.cjks,
+    wikitext.revision.entities,
+    wikitext.revision.urls,
+    wikitext.revision.words,
+    wikitext.revision.uppercase_words,
+    wikitext.revision.punctuations,
+    wikitext.revision.breaks,
+    wikitext.revision.longest_token,
+    wikitext.revision.longest_word,
+    wikitext.revision.numbers / max(wikitext.revision.tokens, 1),
+    wikitext.revision.whitespaces / max(wikitext.revision.tokens, 1),
+    wikitext.revision.markups / max(wikitext.revision.tokens, 1),
+    wikitext.revision.cjks / max(wikitext.revision.tokens, 1),
+    wikitext.revision.entities / max(wikitext.revision.tokens, 1),
+    wikitext.revision.urls / max(wikitext.revision.tokens, 1),
+    wikitext.revision.words / max(wikitext.revision.tokens, 1),
+    wikitext.revision.uppercase_words / max(wikitext.revision.tokens, 1),
+    wikitext.revision.punctuations / max(wikitext.revision.tokens, 1),
+    wikitext.revision.breaks / max(wikitext.revision.tokens, 1),
+    wikitext.revision.longest_token / max(wikitext.revision.tokens, 1),
+    wikitext.revision.longest_word / max(wikitext.revision.tokens, 1)
+]
+
+parse_based = [
+    wikitext.revision.content_chars,
+    wikitext.revision.headings,
+    wikitext.revision.external_links,
+    wikitext.revision.wikilinks,
+    wikitext.revision.tags,
+    wikitext.revision.ref_tags,
+    wikitext.revision.templates,
+    wikitext.revision.content_chars / max(wikitext.revision.tokens, 1),
+    wikitext.revision.headings / max(wikitext.revision.tokens, 1),
+    wikitext.revision.external_links / max(wikitext.revision.tokens, 1),
+    wikitext.revision.wikilinks / max(wikitext.revision.tokens, 1),
+    wikitext.revision.tags / max(wikitext.revision.tokens, 1),
+    wikitext.revision.ref_tags / max(wikitext.revision.tokens, 1),
+    wikitext.revision.templates / max(wikitext.revision.tokens, 1)
+]
+
+badwords = [
+    portuguese.badwords.revision.matches,
+    portuguese.badwords.revision.matches / max(wikitext.revision.words, 1)
+]
+
+informals = [
+    portuguese.informals.revision.matches,
+    portuguese.informals.revision.matches / max(wikitext.revision.words, 1)
+]
+
+dict_words = [
+    portuguese.dictionary.revision.dict_words,
+    portuguese.dictionary.revision.non_dict_words,
+    portuguese.dictionary.revision.dict_words /
+    max(wikitext.revision.words, 1),
+    (portuguese.dictionary.revision.non_dict_words /
+     max(wikitext.revision.words, 1))
+]
+
+# Templates
+infobox_templates = wikitext.revision.template_names_matching(
+    r"Info", name="ptwiki.revision.infobox_templates")
+
+CN_TEMPLATES = [
+    r"Citation[_ ]needed",
+    r"Cn",
+    r"Fact"
+]
```

Comment:
These don't look right.  Maybe grab what we have in articlequality and add it here.